api/envoy/config/route/v3/route_components.proto

```diff
@@ -1629,6 +1629,24 @@ message RateLimit {
       string default_value = 3;
     }
 
+    // The following descriptor entry is appended when the route entry metadata contains a key value:
+    //
+    // .. code-block:: cpp
+    //
+    //   ("<descriptor_key>", "<value_queried_from_route_entry_metadata>")
+    message RouteEntryMetaData {
+      // The key to use in the descriptor entry.
+      string descriptor_key = 1 [(validate.rules).string = {min_len: 1}];
+
+      // Metadata struct that defines the key and path to retrieve the string value. A match will
+      // only happen if the value in the filter metadata is of type string.
```

Comment:
can we add links to other parts of the docs that explain the different sources of metadata here and above?

Reply:
The problem is, that only dynamic metadata has an explanation: https://www.envoyproxy.io/docs/envoy/latest/configuration/advanced/well_known_dynamic_metadata
I could not find a place in the documentation where the different sources of metadata and their differences are clarified. See: #12556

Reply:
I added the link above for dynamic metadata and https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto.html#envoy-v3-api-field-config-route-v3-route-metadata for the route entry metadata.